frontend/src/routes/__root.tsx

```diff
@@ -4,18 +4,142 @@ import { Logo } from "~/components/logo";
+      <Navigation />
       <Outlet />
-      <TailwindIndicator />
-      <TanStackRouterDevtools />
+      <Footer />
     </main>
   ),
   notFoundComponent: NotFound
 });
 
+function Navigation() {
+  const query = useAuthUser();
+  const user = query.data?.data?.user;
+  if (!user) {
+    return null;
+  }
+  return (
+    <>
+      <header className="flex px-6 border-b border-opacity-65 border-border py-2 items-center bg-toogle t justify-between gap-4">
+        <Logo className="w-10 flex-none h-10 mr-8" />
+        <div className="flex  w-full items-center">
+          <Menubar className="border-none w-fit text-black bg-primary">
+            <MenubarMenu>
+              <MenubarTrigger className="flex justify-center text-sm items-center gap-1">
+                Create
+                <ChevronsUpDown className="w-4" />
+              </MenubarTrigger>
+              <MenubarContent className="border min-w-6 border-border">
+                <MenubarItem className="flex gap-5">
+                  <Folder className="w-4 opacity-50" />
+                  Project
+                </MenubarItem>
+                <MenubarItem className="flex gap-5">
+                  <Globe className="w-4 opacity-50" />
+                  Web Service
+                </MenubarItem>
+                <MenubarItem className="flex gap-5">
+                  <Hammer className="w-4 opacity-50" />
+                  Worker
+                </MenubarItem>
+                <MenubarItem className="flex gap-5">
+                  <AlarmCheck className="w-4 opacity-50" />
+                  CRON
+                </MenubarItem>
+              </MenubarContent>
+            </MenubarMenu>
+          </Menubar>
+          <Search className="relative left-10" />
+          <Input
+            className="px-14 my-1 focus-visible:right-0"
+            placeholder="Search for Service, Worker, CRON, etc..."
+          />
+          <HelpCircle className="w-16 stroke-[1.5px] opacity-70" />
+        </div>
+
+        <Menubar className="border-none w-fit">
+          <MenubarMenu>
+            <MenubarTrigger className="flex justify-center items-center gap-2">
+              <CircleUser className="w-5 opacity-70" />
+              <p>{user.username}</p>
+              <ChevronDown className="w-4 my-auto" />
+            </MenubarTrigger>
+            <MenubarContent className="border min-w-0 mx-9  border-border">
+              <MenubarItem className="flex gap-5">
+                <Settings className="w-4 opacity-50" />
+                Settings
+              </MenubarItem>
+              <MenubarItem className="flex gap-5">
+                <LogOut className="w-4 opacity-50" />
+                Logout
+              </MenubarItem>
+            </MenubarContent>
+          </MenubarMenu>
+        </Menubar>
+      </header>
+    </>
+  );
+}
+
+const linksIconWidth = 15;
+const links = [
+  { name: "Feedback", url: "", icon: <Send width={linksIconWidth} /> },
```

Comment:
```suggestion
  { name: "Feedback", url: "", icon: <Send className="w-4 h-4 flex-none" /> },
```
Use classes for the width instead of a fixed width. 

Also you might want to use the non-rendered component for the item instead to not repeat yourself, like this :

```tsx
const links = [
  { name: "Feedback", url: "", icon: Send },
 { name: "Docs", url: "", icon: BookOpen },
 // etc...
]
```

This way you can pass the same className to all the icons so that they look the same : 

```tsx
{links.map(({...link, icon: Icon }) => (
          <a className="flex underline items-center gap-2" href="">
              {/*...rest of the items */}
              <Icon className="w-4 h-4 flex-none" />
        </a>
)}
```

> [!NOTE]  
> It is always important to add the `flex-none` className to icons because SVG icons tend to get squashed when their flex parent get resized.